### PR TITLE
Issue 2660

### PR DESCRIPTION
**Pull request by gsatimos**

### CODE REVIEW

File: web-app/js/portal/ui/openlayers/control/SpatialConstraint.js [hunk 1]
```diff
@@ -65,11 +67,131 @@ Portal.ui.openlayers.control.SpatialConstraint = Ext.extend(OpenLayers.Control.D
...
+            geometry.components[0].components.splice(i,1);
+        });
+
+        return geometry;
+    },
+
+    _shiftMapCentre: function(lon) {
+        var center = this.map.getCenter();
+
+        if ((center.lon > 0 && lon < 0) || (center.lon < 0 && lon > 0)) {
```
Comment (anguss00) [inline web-app/js/portal/ui/openlayers/control/SpatialConstraint.js]: This function is doing two separate things here - checking if there is a cross over the antimeridian for the centre and a supplied lon value, and then returning a bool if the centre has moved. Maybe pull out the comparo into a separate method, and then keep this function as a straight shift. Better from a testing point of view

File: web-app/js/portal/ui/openlayers/control/SpatialConstraint.js [hunk 2]
```diff
@@ -65,11 +67,126 @@ Portal.ui.openlayers.control.SpatialConstraint = Ext.extend(OpenLayers.Control.D
...
+
+    _polygonLastCoord: function() {
+        return this.polygonCoords[this.polygonCoords.length-1]
+    },
+
+    _polygonCoordsExist: function() {
+        return this.polygonCoords.length >= 1;
+    },
+
+    _antimeridianPointCross: function(lon1, lon2) {
```
Comment (pmbohm) [inline web-app/js/portal/ui/openlayers/control/SpatialConstraint.js]: could you rename this to __polygonAntimeridianPointCross

Comment (gsatimos) [reply web-app/js/portal/ui/openlayers/control/SpatialConstraint.js]: thanks @pmbohm that makes good sense. I'll wait to see if you have more suggestions and then fix it up

File: web-app/js/portal/ui/openlayers/control/SpatialConstraint.js [hunk 3]
```diff
@@ -65,11 +67,126 @@ Portal.ui.openlayers.control.SpatialConstraint = Ext.extend(OpenLayers.Control.D
...
+        return (lonLat1.lat == lonLat2.lat && lonLat1.lon == lonLat2.lon);
+    },
+
+    _polygonRemoveDuplicatePointsGeom: function(geometry) {
+        var lastxy = {x: -1, y: -1};
+        var removeIndex = [];
+
+        var components = geometry.components[0].components;
+        components.forEach(function(component, i) {
+            var thisxy = {x: component.x, y: component.y}
```
Comment (pmbohm) [inline web-app/js/portal/ui/openlayers/control/SpatialConstraint.js]: Could you add semi colons please. Theres some of yours and at least one other

File: src/test/javascript/portal/ui/openlayers/control/SpatialConstraintSpec.js [hunk 4]
```diff
@@ -403,10 +405,10 @@ describe('Portal.ui.openlayers.control.SpatialConstraint', function() {
         it('mousedown method called when map receives mousedown event', function() {
 
             // this mousedown event does call the spatialConstraint's _mapMouseDown method
-            _map.events.triggerEvent("mousedown");
+            map.events.triggerEvent('mousedown');
+
+            expect(polygonConstraint._mapMouseDown).toHaveBeenCalled();
```
Comment (gsatimos) [inline src/test/javascript/portal/ui/openlayers/control/SpatialConstraintSpec.js]: well its much better than what I had. I wouldn't say that it's naive at all

**Comment by pmbohm** [comment]
fixes #2660

**Comment by gsatimos** [comment]
@pmbohm test looks good now. Are you happy to merge? or get a third party (now that we've both touched it)?

**Comment by pmbohm** [comment]
@gsatimos Function rename and semicolons? 
If you fix these then you are OKing my test in a way. I could merge then

**Comment by gsatimos** [comment]
@akashisama would you be in a position to PR this this morning? I wasn't able to answer @pmbohm 's question before I headed off yesterday so it still remains to be PRed

**Pull request merged**